feat: support parallelstore csi driver

DrFaust92 opened the pull request.

apeabody commented:
/gcbrun

apeabody commented on autogen/main/variables.tf.tmpl:
```diff
@@ -838,6 +838,12 @@ variable "stateful_ha" {
+variable "parallelstore_csi_driver" {
+  type        = bool
+  description = "Whether the Parallelstore CSI driver Addon is enabled for this cluster."
+  default     = false
```
As there is a dynamic block using a `for_each`, could the default be made `null` so this is optional?

DrFaust92 replied:
will check

apeabody commented on autogen/main/cluster.tf.tmpl:
```diff
@@ -388,6 +388,14 @@ resource "google_container_cluster" "primary" {
       }
     }
 
+    dynamic "parallelstore_csi_driver_config" {
```
I don't think we need to add for Autopilot clusters as for them it's enabled by default?

DrFaust92 replied:
will check

DrFaust92 replied:
<img width="965" alt="Screenshot 2024-12-16 at 18 18 28" src="https://github.com/user-attachments/assets/a57aae54-9648-4204-ad2a-3f8c28ea8eac" />

Only found this as ref, will go with not setting this on autopilot for now

apeabody reviewed (changes requested):
Thanks for the contribution @DrFaust92!

apeabody commented:
/gcbrun

apeabody commented:
/gcbrun

DrFaust92 commented:
> From the INT test:
> 
> ```
> Error: Null condition
> 
>   on ../../../main.tf line 94, in locals:
>   94:   parallelstore_csi_driver_config = var.parallelstore_csi_driver ? [{ enabled = true }] : []
>     ├────────────────
>     │ var.parallelstore_csi_driver is null
> 
> The condition value is null. Conditions must either be true or false.}
> ```

I think i fixed it now, i addressed a previous comment but didnt address the nullability

apeabody commented:
/gcbrun

apeabody reviewed (approved):
Thanks @DrFaust92!
